### DISCUSSION TITLE
Question about authentication errors

### DISCUSSION THREAD

**Discussion by pedrofontoura** [strapi/strapi#5887] [Legacy Topics]
Just curious, why doesn't strapi handle authentication errors with the standard error response format? For example, an invalid POST /auth/forgot-password response could be:
```json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "This email does not exist."
}
```
Instead, we have:
```json
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": [
        {
            "messages": [
                {
                    "id": "Auth.form.error.user.not-exist",
                    "message": "This email does not exist."
                }
            ]
        }
    ],
    "data": [
        {
            "messages": [
                {
                    "id": "Auth.form.error.user.not-exist",
                    "message": "This email does not exist."
                }
            ]
        }
    ]
}
```

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by derrickmehaffy** [comment]
This is a known issue relating to error messages as they are currently being refactored, this will be fixed before stable. (It generally only happens on plugins for the moment. The Users-permissions plugin is getting a major rewrite so that's likely why this hasn't been fixed yet, easier to just do all the rewrites all at once)
